community/dbms/src/main/java/org/neo4j/commandline/dbms/DiagnosticsReportCommand.java

```diff
@@ -128,6 +129,14 @@ public void execute( String[] stringArgs ) throws IncorrectUsage, CommandFailed
         }
     }
 
+    private String getDefaultFilename() throws UnknownHostException
+    {
+        String hostName = InetAddress.getLocalHost().getHostName();
+        String safeFilename = hostName.replaceAll( "[^a-zA-Z0-9._]+", "_" );
+        SimpleDateFormat dumpFormat = new SimpleDateFormat( "yyyy-MM-dd_HHmmss" );
+        return safeFilename + "-" + dumpFormat.format( new Date() ) + ".zip";
```

Comment:
Possibly better to use Java 8 APIs like `Instant` and `DateTimeFormatter` (thread-safe and can be a constant)